PizzaOrders/PizzaOrders/Program.cs

```diff
@@ -0,0 +1,166 @@
+        {
+            this.pizza = new WhitePizza();
+        }
+        public void BuildDough()
+        {
+            this.pizza.Add("White Dough");
+        }
+        public void BuildSauce()
+        {
+            this.pizza.Add("Tomato Sauce");
+        }
+        public void BuildTopping()
+        {
+            this.pizza.Add("Green Pepper");
+        }
+        public WhitePizza GetProduct()
+        {
+            WhitePizza result = this.pizza;
+            this.Reset();
+            return result;
+        }
+    }
+    public class WhitePizza
+    {
+        private List<object> ingredients = new List<object>();
+        public void Add(string part)
+        {
+            this.ingredients.Add(part);
+        }
+        public string listParts()
+        {
+            string str = "";
+            for (int i = 0; i < this.ingredients.Count; i++)
+            {
+                str += this.ingredients[i] + ",";
+            }
+            return "pizza ingredients:" + str;
+        }
+    }
+
+    public class WheatPizzaBuilder : IBuilder
+    {
+        private WheatPizza pizza = new WheatPizza();
+        public WheatPizzaBuilder()
+        {
+            this.Reset();
+        }
+        public void Reset()
+        {
+            this.pizza = new WheatPizza();
+        }
+        public void BuildDough()
+        {
+            this.pizza.Add("Wheat Dough");
+        }
+        public void BuildSauce()
+        {
+            this.pizza.Add("Tomato Sauce");
+        }
+        public void BuildTopping()
+        {
+            this.pizza.Add("Black Olive");
+        }
+        public WheatPizza GetPizza()
+        {
+            WheatPizza result = this.pizza;
+            this.Reset();
+            return result;
+        }
+    }
+
+    public class WheatPizza
+    {
+        private List<object> ingredients = new List<object>();
+        public void Add(string part)
+        {
+            this.ingredients.Add(part);
+        }
+        public string listIngredients()
+        {
+            string str = "";
+            for (int i = 0; i < this.ingredients.Count; i++)
+            {
+                str += this.ingredients[i] + ",";
+            }
+            return "pizza ingredients:" + str;
+        }
+    }
+
+    public class Director
+    {
+        private IBuilder builder;
+        public IBuilder Builder
+        {
+            set { this.builder = value; }
+        }
+        public void plainPizza()
+        {
+            this.builder.BuildDough();
+            this.builder.BuildSauce();
+        }
+        public void makePizza()
+        {
+            this.builder.BuildDough();
+            this.builder.BuildSauce();
+            this.builder.BuildTopping();
+        }
+    }
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            //WHITE PIZZA:
+            //client
+            var director = new Director();
+            var white = new WhitePizzaBuilder();

```

Comment:
It would be better if you used polymorphism to define builders, without creating new builder variable for each one.